ssl.go

```diff
@@ -50,6 +50,16 @@ func ssl(o values) (func(net.Conn) (net.Conn, error), error) {
 		return nil, fmterrorf(`unsupported sslmode %q; only "require" (default), "verify-full", "verify-ca", and "disable" supported`, mode)
 	}
 
+	// Set Server Name Indication (SNI), if enabled by connection parameters.
+	// By default SNI is on, any value != "1" disables SNI.
+	if sslsni := o["sslsni"]; sslsni == "" || sslsni == "1" {
```

Comment:
It only sets SNI if the first character in the variable is `1`:

https://github.com/postgres/postgres/blob/master/src/interfaces/libpq/fe-secure-openssl.c#L1216

`libpq` also limits the input, too, so you can't use `true` as a value: https://github.com/postgres/postgres/blob/master/src/interfaces/libpq/fe-connect.c#L302-L304

Reply:
> It only sets SNI if the first character in the variable is `1`:
> 
> https://github.com/postgres/postgres/blob/master/src/interfaces/libpq/fe-secure-openssl.c#L1216

Hmm. I wonder if we should have a similarly silly check. `strings.HasPrefix`?

Reply:
> Hmm. I wonder if we should have a similarly silly check. strings.HasPrefix?

I'm not excited about `libpq` behavior around `sslsni` values but makes sense to just replicate `libpq` approach for compatibility reasons. So changed PR to use `strings.HasPrefix`. In the end, I don't think it matters a lot because nobody cares to disable SNI, except for testing purposes.